ftplugin/markdown/quicklink.vim

```diff
@@ -95,4 +95,45 @@ function! OpenLinkOnCurrentLine()
   call system('open ' . expand('<cWORD>'))
 endfunction
 
+function! s:OnMarkdownLink()
+    return synIDattr(synID(line("."),col("."),1),"name") == "markdownLinkText" ||
+                \ synIDattr(synID(line("."),col("."),1),"name") == "markdownLinkTextDelimiter" ||
+                \ synIDattr(synID(line("."),col("."),1),"name") == "mkdLink" ||
+                \ synIDattr(synID(line("."),col("."),1),"name") == "mkdDelimiter"
```

Comment:
I think you can simplify and clarify this by saving off the `synIDattr` return value and using a list count check:

``` vim
" Define this as a constant(ish) near the top of the script
let s:MARKDOWN_LINK_SYNTAX_IDS = [
  \ "markdownLinkText",
  \ "markdownLinkTextDelimiter",
  \ "mkdLink",
  \ "mkdDelimiter"
  \ ]

...

function! OnMarkdownLink()
  let current_syntax_id = synIDattr(synID(line("."), col("."), 1), "name")
  return count(s:MARKDOWN_LINK_SYNTAX_IDS, current_syntax_id) != 0
endfunction
```